federation/pkg/kubefed/init/init.go

```diff
@@ -167,11 +171,19 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 	}
 
 	// 2. Expose a network endpoint for the federation API server
-	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName, dryRun)
+	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName, dryRun, cloudEnvironment)
 	if err != nil {
 		return err
 	}
-	ips, hostnames, err := waitForLoadBalancerAddress(hostClientset, svc, dryRun)
+	ips := []string{}
+	hostnames := []string{}
+	if !dryRun {
+		if cloudEnvironment {
+			ips, hostnames, err = util.WaitForLoadBalancerAddress(hostClientset, svc)
+		} else {
+			ips, err = util.GetFirstNodeIP(hostClientset)
+		}
```

Comment:
Would be preferable to have this ip be configurable and not only auto-magically devined.  There are use cases where you might have more than 1 "external" ip and need to provide the specific one for this service.